src/main/java/org/elasticsearch/search/highlight/Highlighter.java

```diff
@@ -26,4 +28,6 @@
     String[] names();
 
     HighlightField highlight(HighlighterContext highlighterContext);
+
+    public boolean cannotHighlight(FieldMapper fieldMapper);
```

Comment:
I find the negation a bit weird, but I see why you went for it. Still, last chance to change it to `canHighlight` , would it be bad?